### PR TITLE
[MXNET-690] Add tests for initializers in R

### PR DESCRIPTION
**Pull request by anirudhacharya**
## Description ##
Add tests for uniform, normal and Xavier initializers.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

## Comments
- @ankkhedia @hetong007

### CODE REVIEW

File: R-package/tests/testthat/test_initializer.R [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        weight_mean <- mean(as.array(X_weight))
+        expect_equal(weight_mean, 0, tolerance=1e-2)
+      }
+    }
+  }
+})
+
+testthat("mx.init.internal.default", {
+  sample_bias <- mxnet:::mx.init.internal.default("X_bias", c(5, 10, 100),
+                                          ctx = mx.ctx.default())
```
Comment (hetong007) [inline R-package/tests/testthat/test_initializer.R]: Incorrect indent.

File: R-package/tests/testthat/test_initializer.R [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  expect_equal(weight_mean, 0, tolerance=1e-2)
+  expect_equal(weight_sd, 0.1, tolerance=1e-2)
+})
+
+testthat("mx.init.Xavier", {
+  xavier_init <- mx.init.Xavier()
+  expect_equal(typeof(xavier_init), 'closure')
+
+  # default parameters
+  shape <- c(2, 3, 324, 324)
```
Comment (hetong007) [inline R-package/tests/testthat/test_initializer.R]: Is there a reason to pick `324`?

Comment (anirudhacharya) [reply R-package/tests/testthat/test_initializer.R]: no particular reason, I just did not want to pick a number too low that the test becomes flaky or too high that the test takes very long to execute. It was quite random

Comment (hetong007) [reply R-package/tests/testthat/test_initializer.R]: OK. As long as it works.

**Comment by anirudhacharya** [comment]
@hetong007 ran `formatR` on the whole `tests` folder

**Comment by hetong007** [comment]
LGTM

**Pull request merged**